text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+- Downstream components should be able to engage with the Entity model in Resource.
+
+The following changes are made:
+
+### Resource
+
+| Field | Type | Description | Changes |
+| ----- | ---- | ----------- | ------- |
+| schema_url | string | The Schema URL, if known. This is the identifier of the Schema that the resource data  is recorded in.  This field is deprecated and should no longer be used. | Will be deprecated |
+| dropped_attributes_count |  integer | dropped_attributes_count is the number of dropped attributes. If the value is 0, then no attributes were dropped. | Unchanged |
+| attributes | repeated KeyValue | Set of attributes that describe the resource.<br/><br/>Attribute keys MUST be unique (it is not allowed to have more than one attribute with the same key).| Unchanged |
+| entities | repeated ResourceEntityRef | Set of entities that participate in this Resource. | Added |
+
+The DataModel would ensure that attributes in Resource are produced from both the identifying and descriptive attributes of Entity.  This does not mean the protocol needs to transmit duplicate data, that design is TBD.
+
+### ResourceEntityRef
+
+The entityref data model, would have the following changes from the original [entity OTEP](https://github.com/open-telemetry/oteps/blob/main/text/entities/0256-entities-data-model.md) to denote references within Resource:
+
+| Field | Type | Description | Changes |
+| ----- | ---- | ----------- | ------- |
+| schema_url | string | The Schema URL, if known. This is the identifier of the Schema that the entity data  is recorded in. To learn more about Schema URL ([see docs](https://opentelemetry.io/docs/specs/otel/schemas/#schema-url)) | added |
+| type | string | Defines the type of the entity. MUST not change during the lifetime of the entity. For example: "service" or "host". This field is required and MUST not be empty for valid entities. | unchanged |
+| identifying_attributes_keys | repeated string | Attribute Keys that identify the entity.<br/>MUST not change during the lifetime of the entity. The Id must contain at least one attribute.<br/><br/>These keys MUST exists in Resource.attributes.<br/><br/>Follows OpenTelemetry common attribute definition. SHOULD follow OpenTelemetry semantic conventions for attributes.| now a reference |
+| descriptive_attributes_keys | repeated string | Descriptive (non-identifying) attribute keys of the entity.<br/>MAY change over the lifetime of the entity. MAY be empty. These attribute keys are not part of entity's identity.<br/><br/>These keys MUST exist in Resource.attributes.<br/><br/>Follows any value definition in the OpenTelemetry spec - it can be a scalar value, byte array, an array or map of values. Arbitrary deep nesting of values for arrays and maps is allowed.<br/><br/>SHOULD follow OpenTelemetry semantic conventions for attributes.| now a reference |
```

Comment:
If an entity's descriptive attributes can change over the lifetime of the entity, and the descriptive attributes reference the resource attributes, then either:
- the resource attributes are now subject to change from one export to another
- OR descriptive attributes are static, but can change via entity lifecycle events

Is that how you see it?

Reply:
I think the nuance of allowing resource changes in an SDK will need to be in a follow up OTEP expanding on that component, likely from client-SIG.  For now - I'd like to make progress on the simple merge algorithm and move from this proposal to prototypes of the merge algorithm and show casing how it interacts with existing resource detection in various languages.

Reply:
Resolving as we will have a follow on OTEP to address this, as described in the OTEP.